yfinance/base.py

```diff
@@ -589,20 +599,20 @@ def get_earnings_dates(self, limit=12, proxy=None) -> Optional[pd.DataFrame]:
 
         # Parse earnings date string
         cn = "Earnings Date"
-        # - remove AM/PM and timezone from date string
-        tzinfo = dates[cn].str.extract('([AP]M[a-zA-Z]*)$')
-        dates[cn] = dates[cn].replace(' [AP]M[a-zA-Z]*$', '', regex=True)
-        # - split AM/PM from timezone
-        tzinfo = tzinfo[0].str.extract('([AP]M)([a-zA-Z]*)', expand=True)
-        tzinfo.columns = ["AM/PM", "TZ"]
-        # - combine and parse
-        dates[cn] = dates[cn] + ' ' + tzinfo["AM/PM"]
-        dates[cn] = pd.to_datetime(dates[cn], format="%b %d, %Y, %I %p")
-        # - instead of attempting decoding of ambiguous timezone abbreviation, just use 'info':
-        self._quote.proxy = proxy or self.proxy
-        tz = self._get_ticker_tz(proxy=proxy, timeout=30)
-        dates[cn] = dates[cn].dt.tz_localize(tz)
 
+        def map_date(time_str: str):
+            tz_match = re.search('([AP]M)([a-zA-Z]*)$', time_str)
+            tz_str = tz_match.group(2).strip()
+            # - remove AM/PM and timezone from date string
+            time_str = time_str.replace(tz_str, "")
+            try:
+                tz = pytz.timezone(tz_str)
+            except pytz.UnknownTimeZoneError:
+                tz = get_ticker_tz()
```

Comment:
If `get_ticker_tz()` is returning None then the ticker isn't listed. So the returned earnings dates for legitimate tickers should already be localized.

Reply:
As far as I remember, timezones like „EDT“ are unknown to Python. Let me point that out in a dedicated test.

Reply:
Exactly, that's why you shouldn't use `tz_str`.

Reply:
But the timezones from the earnings list can differ from the ticker itself. I would prefer them and use the general ticker tz as fallback.

Reply:
Now I understand your position. I've pushed a commit to improve your proposal, easier than discussing here, can always rollback.